modules/highgui/src/window_w32.cpp

```diff
@@ -1421,7 +1422,20 @@ MainWindowProc( HWND hwnd, UINT uMsg, WPARAM wParam, LPARAM lParam )
           GetClientRect( window->hwnd, &rect );
 
           SIZE size = {0,0};
-          icvGetBitmapData( window, &size, 0, 0 );
+#ifdef HAVE_OPENGL
+          if (window->useGl)
```

Comment:
What about moving this if-block into `icvGetBitmapData`?

Reply:
not my call either way. I chose not to because I don't fully comprehend the naming convention of `icvGetBitmapData` and related functions and they seem to not deal with OpenGL at all.

I would suggest not querying whatever bitmap is stored for whatever mode, but storing the size in `window` explicitly when bitmap data is set. then, none of these case distinctions have to be made.